Added coverage for the Tor domains of Reddit, Twitter, and Facebook

DandelionSprout opened the pull request:
# Creating the pull request

> Please include a summary of the change and which issue is fixed\
> If the related issue does not exist in our repository, please create it before making pull request\
> It is highly recommended to use our [Web Reporting Tool](https://kb.adguard.com/en/technical-support/reporting-tool) instead of creating an issue on GitHub directly\
> Please note, that we verify every pull request manually, so it may take time to apply it

## Prerequisites

### To avoid invalid pull requests, please check and confirm following terms

- [ ] This is not an ad/bug report;
- [x] My code follows the [guidelines](https://github.com/AdguardTeam/AdguardFilters/blob/master/CONTRIBUTING.md) and [syntax](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) of this project;
- [x] I have performed a self-review of my own changes;
- [x] My changes do not break web sites, apps and files structure.

## What problem does the pull request fix?

### If the problem does not fall under any category that is listed here, please write a comment below in corresponding section

- [x] Missed ads or ad leftovers;
- [ ] Website or app doesn't work properly;
- [ ] AdGuard gets detected on a website;
- [ ] Missed analytics or tracker;
- [ ] Social media buttons — share, like, tweet, etc;
- [x] Annoyances — pop-ups, cookie warnings, etc;
- [x] Filters maintenance.

## What issue is being fixed?

Coverage of major social medias' Tor addresses had not been universally applied in Base Filter and Popups Filter, primarily in the case of Reddit whose Tor address is the newest created.

### Enter the issue address

N/A

### Add your comment and screenshots

#### If possible, a screenshot of a page or application should not be cropped too much. Otherwise, it is not always clear where the element is located

1. Your comment

Although the `[$` syntax was new to me, I went through the 3 most important files in the repo to modify their existing entries to cover the Tor addresses.

2. Screenshots

<details><summary>Reddit Onion site ad currently not blocked by big adblocker lists:</summary>

![image](https://user-images.githubusercontent.com/22780683/221485786-ea2f8d57-7464-4662-8628-3dd414901c9f.png)

</details>

### Terms

- [x] By submitting this issue, I agree that pull request does not contain private info and all conditions are met

DandelionSprout committed:
Added the Tor domains for Reddit and Facebook

DandelionSprout committed:
Added the Tor domains for Reddit/FB/Twitter

DandelionSprout committed:
Added the Tor domains for Reddit and Facebook

krystian3w commented on BaseFilter/sections/css_extended.txt:
```diff
@@ -2424,8 +2424,8 @@ yeptube.com#?#.wrapper > div.container:has(> div.c-title > div.c-medtitle-output
 ! Facebook Marketplace
-[$path=/marketplace/]facebook.com#$?#div[class][style^="max-width:"]:has(> span > div  a[href^="/ads/about/"]) { display: none !important; }
-[$path=/marketplace/]facebook.com#$?#div[class][style^="max-width:"]:has(> span > div > a[target="_blank"]:not([href^="/marketplace/"])) { display: none !important; }
+[$path=/marketplace/]facebook.com,facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion#$?#div[class][style^="max-width:"]:has(> span > div  a[href^="/ads/about/"]) { display: none !important; }
```
These better write as new line if nothing has been changed to use:

`[$path=/marketplace/]facebook.com,[$path=/marketplace/]facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion`

DandelionSprout replied:
You sound like you know more about `$path` than I do. Give me 20min.

krystian3w replied:
Recently, SKEIDs combined identical rules and somewhere with the help of path was narrowed down and had to ditch it, however.

> https://github.com/AdguardTeam/AdguardFilters/pull/137586#discussion_r1051386265

So it would be useful to document how the path affects decimal domains.

It may also be about backward compatibility so that the simpler filter does not deactivate in a non-updated application/add-on.

krystian3w commented:
Also look as legacy short names, due 2 months ago Alex-302 don't changed all: https://github.com/AdguardTeam/AdguardFilters/commit/dcc4e39e3433f4542889b990fc163fa55ed78327

https://github.com/AdguardTeam/AdguardFilters/issues/137674#issuecomment-1361259557

DandelionSprout committed:
Improving coverage for Facebook's Tor domain

DandelionSprout committed:
Improving coverage of Facebook's Tor domain

DandelionSprout committed:
Improving coverage of Facebook's Tor domain

It is however unclear to me why these entries are in AdGuard Russian Filter and not in Base.

krystian3w commented:
I see these legacy domains in files: `facebookcorewwi.onion`

https://github.com/AdguardTeam/AdguardFilters/blob/4d3d8e2c41a94156b82b54432b0319da1aa1d47a/AnnoyancesFilter/Other/sections/annoyances.txt#L2175-L2176

https://github.com/AdguardTeam/AdguardFilters/blob/4d3d8e2c41a94156b82b54432b0319da1aa1d47a/AnnoyancesFilter/Other/sections/annoyances.txt#L2223-L2224

based by dumb regex: `/(^|\,|\|)[\w]{1,16}\.onion/`

So not all was changed in: https://github.com/AdguardTeam/AdguardFilters/commit/dcc4e39e3433f4542889b990fc163fa55ed78327 (https://github.com/AdguardTeam/AdguardFilters/issues/137674#issuecomment-1361259557)

DandelionSprout commented:
Since `facebookcorewwwi.onion` is supposed to be spelt with 3 Ws instead of 2, that explains why it took me a while to notice those remnants. I'll make sure to remove and fix them.

krystian3w commented:
Other "too short" domains to check as still live:

* `nytimes3xbfgragh.onion` "nytimes com"

* <details><summary>NSFW</summary><code>pornhubthbh7ap3u.onion</code></details>

* `3g2upl4pq6kufc4m.onion` "duck duck go"

Can be moved to other PR to avoid retitle these.

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Increased coverage of Tor domains

DandelionSprout committed:
Increased coverage of Tor domains

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Improved coverage of Tor domains

DandelionSprout committed:
Addressing syntax feedback

DandelionSprout committed:
Addressing syntax feedback

DandelionSprout commented:
Should be ready to consider for merging now.

krystian3w commented:
I see path cover all domains separated by comma:

```adb
[$path=/^\\/$/]deepl.com,translate.google.com##body > *
[$path=/^(/en/translator|\\/)$/]deepl.com,translate.google.com##body > *
```

Also these is applied only on first:
```
deepl.com,[$path=/^\\/$/]translate.google.com##body > *
```

So first was correct:

> `[$path=/marketplace/]facebook.com,facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion#$?#div[class][style^="max-width:"]:has(> span > div  a[href^="/ads/about/"]) { display: none !important; }`

krystian3w commented on BaseFilter/sections/css_extended.txt:
```diff
@@ -2424,8 +2424,8 @@ yeptube.com#?#.wrapper > div.container:has(> div.c-title > div.c-medtitle-output
 ! Facebook Marketplace
-[$path=/marketplace/]facebook.com#$?#div[class][style^="max-width:"]:has(> span > div  a[href^="/ads/about/"]) { display: none !important; }
-[$path=/marketplace/]facebook.com#$?#div[class][style^="max-width:"]:has(> span > div > a[target="_blank"]:not([href^="/marketplace/"])) { display: none !important; }
+[$path=/marketplace/]facebook.com,[$path=/marketplace/]facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion#$?#div[class][style^="max-width:"]:has(> span > div  a[href^="/ads/about/"]) { display: none !important; }
```
`$path` don't work after duplicate :( - first is used on all listed domains.

DandelionSprout replied:
At that point, I think it's better to wait to hear from AdGuard devs about what `$path` is and how it works like.

Alex-302 replied:
Incorrect rule.
Correct - with one modifier:

```adblock
[$path=/marketplace/]facebook.com,facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion#$?#div[class][style^="max-width:"]:has(> span > div  a[href^="/ads/about/"]) { display: none !important; }
```

![image](https://user-images.githubusercontent.com/8361299/222120711-e5e71c1f-730a-4db9-883b-a25a19b55e6d.png)

Alex-302 reviewed (changes requested):
* Please revert all `#%#` JS rules (non-scriptlets). After change, that rules will stop work in Firefox extension.

* New domains for `#` rules must be added at the beginning of the domains list (you can leave as is for this time).

DandelionSprout commented:
Thanks for the clarifications. I'll see to which extent I can implement them overnight.

DandelionSprout committed:
Split up a non-scriptlet #%# entry

DandelionSprout committed:
Corrected 2 non-scriptlet #%# entries

DandelionSprout committed:
Corrected more entries

DandelionSprout committed:
Corrected 2 $path entries

DandelionSprout committed:
Corrected 2 $path entries

DandelionSprout commented:
I think I've fixed it now.

krystian3w commented:
> that rules will stop work in Firefox extension.

Opera have similar limitations, but also reject all rule due joining domains?
